mitosheet/src/plugin.tsx

```diff
@@ -198,6 +214,76 @@ function activateMitosheetExtension(
         }
     })
 
+    app.commands.addCommand('mitosheet:write-generated-code-cell-by-id', {
+        label: 'Writes the generated code for a deafult dataframe output mitosheet. Writes the code to the code cell below the specified code cell',
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        execute: (args: any) => {
+            const analysisName = args.analysisName as string;
+            const codeLines = args.code as string[];
+            const telemetryEnabled = args.telemetryEnabled as boolean;
+            const publicInterfaceVersion = args.publicInterfaceVersion as PublicInterfaceVersion;
+            const inputCellExecutionCount = args.inputCellExecutionCount as number | undefined; 
+
+            // This is the last saved analysis' code, which we use to check if the user has changed
+            // the code in the cell. If they have, we don't want to overwrite their changes automatically.
+            const oldCode = args.oldCode as string[];
+            const newCode = getCodeString(analysisName, codeLines, telemetryEnabled, publicInterfaceVersion, true);
+            const notebook = notebookTracker.currentWidget?.content;
+            const cells = notebook?.model?.cells;
+
+            if (inputCellExecutionCount === undefined || notebook === undefined || cells === undefined) {
+                return;
+            }
+
+            const mimeRenderInputCellIndex = getCellIndexByExecutionCount(cells, inputCellExecutionCount);
+            console.log('mimeRenderInputCellIndex', mimeRenderInputCellIndex)
+            if (mimeRenderInputCellIndex === undefined) {
+                // If the code cell that created the mitosheet mime render does not exist, 
+                // just return. I don't think this should ever happen because you can't 
+                // have a mimerender for a code cell that does not exist anymore.
+                return;
+            }
+
+            let codeCell = getCellAtIndex(cells, mimeRenderInputCellIndex + 1)
+            const codeCellText = getCellText(codeCell);
+
+            if (codeCell === undefined) {
```

Comment:
Turn this logic into a helper function and share it with the other write generated code